app/controllers/search_stats_controller.rb

```diff
@@ -3,6 +3,10 @@
 class SearchStatsController < ApplicationController
   # GET /search_stats
   def index
-    @pagy, @search_stats = pagy(SearchStat.all)
+    @pagy, @search_stats = pagy(current_user.search_stats)
+  end
+
+  def show
+    @search_stat = SearchStat.includes(:result_links).find(params[:id])
```

Comment:
Authorization 😉 The user can provide the `id` of other's search stat.

```suggestion
    @search_stat = current_user.search_stats.includes(:result_links).find(params[:id])
```